### ISSUE TITLE
The provided rpm doesn't work in Fedora 35

### ISSUE THREAD

**Issue by sunjay140** [ollm/OpenComic#130]
Version tested - 0.2.1

```
sakura@fedora ~> sudo dnf install https://github.com/ollm/OpenComic/releases/download/v0.2.1/opencomic-0.2.1.x86_64.rpm
[sudo] password for sakura: 
Last metadata expiration check: 0:39:43 ago on Wed 30 Mar 2022 08:27:01 PM AST.
opencomic-0.2.1.x86_64.rpm                                                                                                            13 MB/s |  69 MB     00:05    
Error: 
 Problem: conflicting requests
  - nothing provides gconf-service needed by opencomic-0.2.1-1.x86_64
  - nothing provides gconf2 needed by opencomic-0.2.1-1.x86_64
  - nothing provides libappindicator1 needed by opencomic-0.2.1-1.x86_64
  - nothing provides libnotify4 needed by opencomic-0.2.1-1.x86_64
  - nothing provides libnss3 needed by opencomic-0.2.1-1.x86_64
  - nothing provides libxtst6 needed by opencomic-0.2.1-1.x86_64
(try to add '--skip-broken' to skip uninstallable packages)
````

**Comment by zpangwin** [comment]
Still doesn't work in Fedora but I got a little farther.

After downloading rpm:

    wget https://github.com/ollm/OpenComic/releases/download/v0.2.1/opencomic-0.2.1.x86_64.rpm

Had initially tried to install manually using `dnf` and got the following error:

    # dnf install ./opencomic-0.2.1.x86_64.rpm 
    Last metadata expiration check: 3:37:38 ago on Thu 27 Oct 2022 07:23:58 PM EDT.
    Error: 
     Problem: conflicting requests
      - nothing provides libnotify4 needed by opencomic-0.2.1-1.x86_64
      - nothing provides gconf-service needed by opencomic-0.2.1-1.x86_64
      - nothing provides gconf2 needed by opencomic-0.2.1-1.x86_64
      - nothing provides libappindicator1 needed by opencomic-0.2.1-1.x86_64
      - nothing provides libnss3 needed by opencomic-0.2.1-1.x86_64
      - nothing provides libxtst6 needed by opencomic-0.2.1-1.x86_64
    (try to add '--skip-broken' to skip uninstallable packages

I noticed it seemed to be referring to packages based on Debian or some other naming standard instead of what Fedora calls them. Since I was just trying to install and didn't need the dev libraries (e.g. `-devel`) version of the packages, I used the below command to make sure all of the appropriate packages would be there (after some research to confirm equivalent package names and the gconf-service - technically `org.gnome.GConf.service` - was included in the `GConf`2 package)

Install dependencies

    # dnf install -y libnotify GConf2 libappindicator nss libXtst

At this point, I tried the recommended (per `dnf`'s error message):

    # dnf install --skip-broken ./opencomic-0.2.1.x86_64.rpm

But that did a fat lot of nothing. So next, I tried falling back to the `rpm` tool itself. That seemed to install, but I did get an error about mimetypes:

    # rpm --install --nodeps ./opencomic-0.2.1.x86_64.rpm 
    Error in file "/usr/share/applications/krita_jpeg.desktop": "jpeg/jfif" is an invalid MIME type ("jpeg" is an unregistered media type)
     
    # cd /usr/share/applications
    # ls -acl | grep -i opencomic
    -rw-rw-r--. 1 root root   381 Oct 27 23:42 opencomic.desktop
     
     # grep Exec opencomic.desktop 
     Exec=/opt/OpenComic/opencomic %U
     
    # ls -acl /opt | grep -i opencomic
    drwxr-xr-x. 1 root root      574 Oct 27 23:42 OpenComic
     
    # ls -acl /opt/OpenComic/
    total 175816
    drwxr-xr-x. 1 root root       574 Oct 27 23:42 .
    drwxr-xr-x. 1 root root       172 Oct 27 23:42 ..
    -rw-r--r--. 1 root root    124662 Oct 27 23:42 chrome_100_percent.pak
    -rw-r--r--. 1 root root    187289 Oct 27 23:42 chrome_200_percent.pak
    -rwsr-xr-x. 1 root root   4746192 Oct 27 23:42 chrome-sandbox
    -rw-r--r--. 1 root root  10528096 Oct 27 23:42 icudtl.dat
    -rwxr-xr-x. 1 root root    249544 Oct 27 23:42 libEGL.so
    -rwxr-xr-x. 1 root root   3053896 Oct 27 23:42 libffmpeg.so
    -rwxr-xr-x. 1 root root   6992080 Oct 27 23:42 libGLESv2.so
    -rwxr-xr-x. 1 root root   3907784 Oct 27 23:42 libvk_swiftshader.so
    -rwxr-xr-x. 1 root root   6712224 Oct 27 23:42 libvulkan.so
    -rw-r--r--. 1 root root      1060 Oct 27 23:42 LICENSE.electron.txt
    -rw-r--r--. 1 root root   4562357 Oct 27 23:42 LICENSES.chromium.html
    drwxr-xr-x. 1 root root       682 Oct 27 23:42 locales
    -rwxr-xr-x. 1 root root 133675040 Oct 27 23:42 opencomic
    drwxr-xr-x. 1 root root        50 Oct 27 23:42 resources
    -rw-r--r--. 1 root root   5035729 Oct 27 23:42 resources.pak
    -rw-r--r--. 1 root root     51449 Oct 27 23:42 snapshot_blob.bin
    drwxr-xr-x. 1 root root        42 Oct 27 23:42 swiftshader
    -rw-r--r--. 1 root root    172276 Oct 27 23:42 v8_context_snapshot.bin
    -rw-r--r--. 1 root root       107 Oct 27 23:42 vk_swiftshader_icd.json

However, when I exited root and tried running it a a normal user, it still didn't work.

    $ /opt/OpenComic/opencomic
    (node:68876) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
    [68876:1028/000351.933685:FATAL:gpu_data_manager_impl_private.cc(445)] GPU process isn't usable. Goodbye.
    Trace/breakpoint trap (core dumped)

The [electron issue mentioned in the error](https://github.com/electron/electron/issues/23506) appears to have been closed but there have not been any new builds of opencomic yet since then. After looking around to see if other electron apps had similar error messages on Fedora and what the workaround might be, [I eventually found one](https://github.com/ebkr/r2modmanPlus/issues/735#issuecomment-1072886083).

Rerunning with:

    $ /opt/OpenComic/opencomic --no-sandbox

I am able to see the GUI and it doesn't unexpected crash or anything. I was also able to open a folder containing png files and view them.

**Comment by zpangwin** [comment]
I don't have time to test compiling right now but I believe the fix for this issue would involve 2 parts.

First, fixing the dependencies mentioned in the rpm which, in theory, *should* be as simple as updating [package.json](https://github.com/ollm/OpenComic/raw/master/package.json), lines 163-172 (as of time of writing) from:

    		"rpm": {
    			"depends": [
    				"gconf2",
    				"gconf-service",
    				"libnotify4",
    				"libappindicator1",
    				"libxtst6",
    				"libnss3",
    				"unrar"
    			]

to

    		"rpm": {
    			"depends": [
    				"GConf2",
    				"libnotify",
    				"libappindicator",
    				"libXtst",
    				"nss",
    				"unrar"
    			]

And the second, would be related to broader `contextIsolation` / `--no-sandbox` issue I mentioned above, which personally I feel would be out of scope for this issue but probably should warrant a separate issue unless it magically disappears with a fresh build.

**Comment by ollm** [comment]
Try this beta version, it has the fix for the dependencies a6f8f004b9f2fff0b366f8b2ac4ecf50895a412b and uses a newer version of electron (Although not the latest), it is possible that both errors are fixed.

https://mega.nz/file/WTYkAS4b#qfjUROJP-E1Vg3DP_xp3PUFIRWDVoo7Jbc3hapVCHfA

**Comment by ollm** [comment]
I'm closing this as probably fixed in [v1.0.0-beta.4](https://github.com/ollm/OpenComic/releases)

Feel free to comment if steel happening.
